### ISSUE TITLE
New rule: report string concatenation that could be a string template

### ISSUE THREAD

**Issue by xouabita** [arturbosch/detekt#2133]
### Non Compliant

```kotlin
val foo = "test = " + bar
```

### Compliant

```kotlin
val foo = "test = $bar"
val baz = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" +
     "bbbbbbbbbbbbbbb"
```

It doesn't report if the concatenation is at the end of the line.

Happy to contribute this one 😃

**Comment by schalkms** [comment]
Thanks for your contribution!
For me, this seems to be a matter of taste.
We try to include contribute rules that are useful for the majority of detekt's user base. I don't think this is the case here, since it doesn't matter which type of style I use. The generated code looks the same for both code snippets.
If you really need this rule for your team, I'd ask you to write a custom domain specific rule [1].
We are happy to help out, if you have any questions.

[1] https://arturbosch.github.io/detekt/extensions.html#custom-rulesets

**Comment by xouabita** [comment]
Ok no problem, we already have this as a custom rule 😄

**Comment by schalkms** [comment]
Nice! I’m happy to hear that the custom rule integration feature works for you.
